### ISSUE TITLE
serverSideTranslations loads all namespaces if given an empty array

### ISSUE THREAD

**Issue by jvcmanke** [isaachinman/next-i18next#1052]
## Describe the bug
When you call `serverSideTranslations` with an empty namespace array all namespaces are loaded.

## Occurs in next-i18next version
8.1.1

## Steps to reproduce
[Reproduction](https://codesandbox.io/s/fervent-edison-xgik9?file=/public/locales/en/ns2.json) -> Check the console

Basically:

```javascript
serverSideTranslations(locale, [])
```

## Expected behaviour
No namespace should be loaded

**Comment by isaachinman** [comment]
I think we'll need to add a check [here](https://github.com/isaachinman/next-i18next/blob/master/src/config/createConfig.ts#L79) for `Array.isArray(combinedConfig.ns)`.

A PR would be very helpful!

**Comment by jvcmanke** [comment]
I actually think the problem is [here](https://github.com/isaachinman/next-i18next/blob/149b592b41c2286aaf04bf5b3a093023ac0722bd/src/serverSideTranslations.ts#L13) and [here](https://github.com/isaachinman/next-i18next/blob/149b592b41c2286aaf04bf5b3a093023ac0722bd/src/serverSideTranslations.ts#L56), we should use `undefined` or `null` as the default for `namespacesRequired` and do:

```typescript
  if (!namespacesRequired) {
    const getAllNamespaces = (path: string) =>
      fs.readdirSync(path)
        .map(file => file.replace(`.${localeExtension}`, ''))

    namespacesRequired = getAllNamespaces(path.resolve(process.cwd(), `${localePath}/${defaultLocale}`))
  } else {
    namespacesRequired = []
  }
```

**Comment by isaachinman** [comment]
@jvcmanke Interested in working on a PR?

**Comment by isaachinman** [comment]
Fixed by #1052.
